### DISCUSSION TITLE
Pagination broke on different $first numbers

### DISCUSSION THREAD

**Discussion by ccong2** [community/community#24402] [Apps, API and Webhooks]
<p>I’m sending the following GraphQL API query. It returns 183 repos. However, it seems I can only set the <strong>‘first’</strong> argument in the search term as 10 and 11, any other number would not work. I’m not sure if is it a known problem, or this number depends on what fields I put in the query?</p>
<pre><code>{
  rateLimit {
    cost
    remaining
    resetAt
  }
  search(query: "license:mit created:2008-03-01..2008-03-31", type: REPOSITORY, first: 11) {
    repositoryCount
    pageInfo {
      endCursor
      startCursor
      hasNextPage
    }
    nodes {
      ... on Repository {
        name
        createdAt
        updatedAt
        defaultBranchRef {
          target {
            ... on Commit {
              history {
                totalCount
                pageInfo {
                  startCursor
                  endCursor
                  hasNextPage
                }
                nodes {
                  author {
                    user {
                      organizations(first: 1) {
                        totalCount
                      }
                    }
                  }
                  additions
                  deletions
                }
              }
            }
          }
        }
      }
    }
  }
}
</code></pre>
<p>Thanks in advance for any responses or suggestions.</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>Thanks for the interesting question!</p>
<p>All GraphQL queries are given an upper bound of ten seconds to execute or a timeout error occurs. Some nodes are more expensive to query than others and will take more time. This means that some complex queries will be closer to the edge and may require restricting the amount of data you request in order for them to execute successfully. In this case, your query can only request the information from 10 or 11 repositories before it goes across the 10s barrier.</p>
<p>I hope that helps!</p>
